javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -797,7 +797,7 @@ private void generateSimpleSyncMethod(ClientMethod clientMethod, JavaType typeBl
             if (CoreUtils.isNullOrEmpty(argumentList)) {
                 // If there are no arguments the argument is Context.NONE
                 argumentList = "Context.NONE";
-            } else if (!clientMethod.getParameters().contains(ClientMethodParameter.CONTEXT_PARAMETER)) {
+            } else if (clientMethod.getParameters().stream().noneMatch(p -> p.getClientType() == ClassType.Context)) {
```

Comment:
I am not sure why we have different ClientMethodParameter.CONTEXT_PARAMETER, but it happens (maybe some code clones the parameter).